process/smartContract/builtInFunctions/saveUserName.go

```diff
@@ -41,6 +41,11 @@ func NewSaveUserNameFunc(
 	return s, nil
 }
 
+// SetNewGasConfig is called whenever gas cost is changed
+func (s *saveUserName) SetNewGasConfig(gasCost *process.GasCost) {
+	s.gasCost = gasCost.BuiltInCost.SaveUserName
```

Comment:
We need to protect `s.gasCost` with a mutex

Reply:
no need. there is no possibility to call the from different go routines

Reply:
protected all the way